src/SwedbankPay.Sdk.Infrastructure/Extensions/UriExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using SwedbankPay.Sdk.Payments;
+using System;
+using System.Collections.Generic;
+
+namespace SwedbankPay.Sdk.Extensions
+{
+    public static class UriExtensions
+    {
+        public static Uri GetUrlWithQueryString(this Uri uri, PaymentExpand paymentExpand)
+        {
+            var paymentExpandQueryString = GetExpandQueryString(paymentExpand);
+            var url = !string.IsNullOrWhiteSpace(paymentExpandQueryString)
+                ? new Uri(uri.OriginalString + paymentExpandQueryString, UriKind.RelativeOrAbsolute)
+                : uri;
+            return url;
+        }
+
+        private static string GetExpandQueryString(PaymentExpand paymentExpand)
+        {
+            var intValue = Convert.ToInt64(paymentExpand);
+            if (intValue == 0)
+                return string.Empty;
+
+            var s = new List<string>();
+            foreach (var enumValue in Enum.GetValues(typeof(PaymentExpand)))
+            {
+                var name = Enum.GetName(typeof(PaymentExpand), enumValue);
+                if (paymentExpand.HasFlag((PaymentExpand)enumValue) && name != "None" && name != "All")

```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)